src/test/scala/chisel3/TypeEquivalenceSpec.scala

```diff
@@ -224,45 +241,94 @@ class TypeEquivalenceSpec extends AnyFlatSpec {
+    DontCare.findFirstTypeMismatch(DontCare, true, true, true) should be(None)
   }
 
   it should "support AsyncReset" in {
-    AsyncReset().findFirstTypeMismatch(AsyncReset(), true, true) should be(None)
-    AsyncReset().findFirstTypeMismatch(Bool(), true, true) should be(
+    AsyncReset().findFirstTypeMismatch(AsyncReset(), true, true, true) should be(None)
+    AsyncReset().findFirstTypeMismatch(Bool(), true, true, true) should be(
       Some(": Left (AsyncReset) and Right (Bool) have different types.")
     )
-    AsyncReset().findFirstTypeMismatch(Clock(), true, true) should be(
+    AsyncReset().findFirstTypeMismatch(Clock(), true, true, true) should be(
       Some(": Left (AsyncReset) and Right (Clock) have different types.")
     )
-    AsyncReset().findFirstTypeMismatch(Reset(), true, true) should be(
+    AsyncReset().findFirstTypeMismatch(Reset(), true, true, true) should be(
       Some(": Left (AsyncReset) and Right (Reset) have different types.")
     )
   }
 
   it should "support Clock" in {
-    Clock().findFirstTypeMismatch(Clock(), true, true) should be(None)
+    Clock().findFirstTypeMismatch(Clock(), true, true, true) should be(None)
   }
 
   it should "support abstract Reset" in {
-    Reset().findFirstTypeMismatch(Reset(), true, true) should be(None)
+    Reset().findFirstTypeMismatch(Reset(), true, true, true) should be(None)
+  }
+
+  it should "support Probe" in {
+    Probe(Bool()).findFirstTypeMismatch(Probe(Bool()), true, true, true) should be(None)
+  }
+
+  it should "support RWProbe" in {
+    RWProbe(Bool()).findFirstTypeMismatch(RWProbe(Bool()), true, true, true) should be(None)
+  }
+
+  it should "detect differences between Probe and Not-Probe" in {
+    Probe(Bool()).findFirstTypeMismatch(Bool(), true, true, true) should be(
+      Some(
+        ": Left (Bool with probeInfo: Some(writeable=false, color=None)) and Right (Bool with probeInfo: (None) have different probeInfo."
+      )
+    )
+  }
+
+  it should "detect differences between probe types" in {
+    RWProbe(Bool()).findFirstTypeMismatch(Probe(Bool()), true, true, true) should be(
+      Some(
+        ": Left (Bool with probeInfo: Some(writeable=true, color=None)) and Right (Bool with probeInfo: (Some(writeable=false, color=None)) have different probeInfo."
+      )
+    )
+  }
+
+  it should "detect differences through probes" in {
+    Probe(Bool()).findFirstTypeMismatch(Probe(Clock()), true, true, true) should be(
+      Some(": Left (Bool) and Right (Clock) have different types.")
+    )
   }
+
+  it should "detect differences in presence of probe colors" in {
+    Probe(Bool()).findFirstTypeMismatch(Probe(Bool(), Green), true, true, true) should be(
+      Some(
+        ": Left (Bool with probeInfo: Some(writeable=false, color=None)) and Right (Bool with probeInfo: (Some(writeable=false, color=Some(Green))) have different probeInfo."
+      )
+    )
+  }
+
+  it should "detect differences in probe colors" in {
```

Comment:
Can this be checked with bundles or some aggregate? Implementation looks like it should work but good to put that under test.

Same bundle/vector, different modifier-s set on an element or similar?

Reply:
ah gotcha. Yeah will add that!

Reply:
added more bundle-y tests in https://github.com/chipsalliance/chisel/pull/4064/commits/373a61dd9eec1a5f7eb583df4e90be8115dd9a04

Reply:
and vector tests too in https://github.com/chipsalliance/chisel/pull/4064/commits/6fd89a362ff3cf78085e50f96ab847e5c219f604